agent/api/task_unix.go

```diff
@@ -15,16 +15,105 @@
 
 package api
 
+import (
+	"path/filepath"
+
+	"github.com/aws/amazon-ecs-agent/agent/config"
+	"github.com/aws/amazon-ecs-agent/agent/utils"
+	"github.com/cihub/seelog"
+	docker "github.com/fsouza/go-dockerclient"
+	specs "github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/pkg/errors"
+)
+
 const (
 	portBindingHostIP = "0.0.0.0"
+	defaultCPUPeriod  = 100000 // 100ms
+	// Reference: http://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_ContainerDefinition.html
+	defaultCPUShare = 2
 )
 
 func (task *Task) adjustForPlatform() {}
 
 func getCanonicalPath(path string) string { return path }
 
-// CgroupEnabled checks whether cgroups need be enabled at the task level
-func (task *Task) CgroupEnabled() bool {
-	//stub for now
-	return false
+// BuildCgroupRoot helps build the task cgroup prefix
+// Example: /ecs/task-id
+func (task *Task) BuildCgroupRoot() (string, error) {
+	taskID, err := task.GetID()
+	if err != nil {
+		return "", errors.Wrapf(err, "task build cgroup root: unable to get task-id")
+	}
+
+	cgroupRoot := filepath.Join(config.DefaultTaskCgroupPrefix, taskID)
+	return cgroupRoot, nil
+}
+
+// BuildLinuxResourceSpec returns a linuxResources object for the task cgroup
+func (task *Task) BuildLinuxResourceSpec() (specs.LinuxResources, error) {
+	linuxResourceSpec := specs.LinuxResources{}
+
+	if !(utils.ZeroOrNil(task.VCPULimit)) {
+		taskCPUPeriod := uint64(defaultCPUPeriod)
+		taskCPUQuota := int64(task.VCPULimit * defaultCPUPeriod)
+
+		// TODO: DefaultCPUPeriod only permits 10VCPUs.
+		// Adaptive calculation of CPUPeriod required for further support
+		linuxResourceSpec.CPU = &specs.LinuxCPU{
+			Quota:  &taskCPUQuota,
+			Period: &taskCPUPeriod,
+		}
+	} else {
+		taskCPUShares := uint64(0)
```

Comment:
I think it would be slightly more idiomatic to write this as `var taskCPUShares uint64`.